services/shhext/service.go

```diff
@@ -251,3 +279,324 @@ func (s *Service) Stop() error {
 	s.tracker.Stop()
 	return nil
 }
+
+// JoinPublicChats join a set of public chats deriving the key from the chat name
+func (s *Service) JoinPublicChats(chats []string) ([]string, error) {
+	var response []string
+
+	for _, chatID := range chats {
+		// Don't allow to join multiple times
+		_, found := s.loadedFilters[chatID]
+		if found {
+			continue
+		}
+
+		symKeyID, err := s.w.AddSymKeyFromPassword(chatID)
+		if err != nil {
+			return nil, err
+		}
+
+		symKey, err := s.w.GetSymKey(symKeyID)
+		if err != nil {
+			return nil, err
+		}
+
+		topic := chat.ToWhisperTopic(chatID)
+		topics := [][]byte{topic[:]}
+
+		filterOpts := &whisper.Filter{
+			KeySym:   symKey,
+			Topics:   topics,
+			AllowP2P: true,
```

Comment:
`AllowP2P` allows to receive p2p messages that can be expired, do we need it here for public chats?

Reply:
oh maybe yes if it's used for mail servers too

Reply:
correct, we use it for mailservers